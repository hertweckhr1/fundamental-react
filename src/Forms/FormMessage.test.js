import FormMessage from './FormMessage';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<Forms />', () => {
    const formMessage = (
        <FormMessage type='help'>
                    Pellentesque metus lacus commodo eget justo ut rutrum varius nunc
        </FormMessage>
    );

    test('create form message item', () => {
        // create form set with form inputs
        let component = renderer.create(formMessage);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the FormMessage component', () => {
            const element = mount(<FormMessage data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
