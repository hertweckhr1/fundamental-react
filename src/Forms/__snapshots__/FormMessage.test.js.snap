// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forms /> create form message item 1`] = `
<span
  className="fd-form__message fd-form__message--help"
>
  Pellentesque metus lacus commodo eget justo ut rutrum varius nunc
</span>
`;
